demo/android/OrcaDemo/orca-demo-app/src/main/java/ai/picovoice/orcademo/MainActivity.java

```diff
@@ -322,25 +350,199 @@ private void stopPlayback() {
         runOnUiThread(() -> setUIState(UIState.EDIT));
     }
 
+    public void onStreamSwitchClick(View view) {
+        if (orca == null) {
+            displayError("Orca is not initialized");
+            streamSwitch.setChecked(false);
+            return;
+        }
+
+        try {
+            if (orcaStream == null) {
+                orcaStream = orca.streamOpen(new OrcaSynthesizeParams.Builder().build());
+                runOnUiThread(() -> {
+                    streamSecsTextView.setText("");
+                    streamSecsTextView.setVisibility(View.VISIBLE);
+                });
+            } else {
+                orcaStream.close();
+                orcaStream = null;
+                runOnUiThread(() -> streamSecsTextView.setVisibility(View.INVISIBLE));
+            }
+        } catch (OrcaException e) {
+            onOrcaException(e);
+        }
+    }
+
     public void onSynthesizeClick(View view) {
         if (orca == null) {
             displayError("Orca is not initialized");
             synthesizeButton.setChecked(false);
             return;
         }
 
-        if (synthesizeButton.isChecked()) {
-            String text = synthesizeEditText.getText().toString();
-            if (!previousText.equals(text)) {
-                runSynthesis(text);
+        String text = synthesizeEditText.getText().toString();
+        if (orcaStream == null) {
+            if (synthesizeButton.isChecked()) {
+                if (!previousText.equals(text)) {
+                    runSynthesis(text);
+                } else {
+                    startPlayback();
+                }
             } else {
-                startPlayback();
+                stopPlayback();
             }
         } else {
-            stopPlayback();
+            runStreamSynthesis(text);
+        }
+    }
+
+    private void stopStreamPlay() {
+        try {
+            runOnUiThread(() -> {
+                setUIState(UIState.EDIT);
+                infoTextView.setText("");
+                streamTextView.setVisibility(View.INVISIBLE);
+                synthesizeEditText.setVisibility(View.VISIBLE);
+            });
```

Comment:
seems like this should all be handled in `setUIState` (which already is wrapped in a `runOnUIThread btw)